factor/email/classes/factor.php

```diff
@@ -31,4 +32,45 @@
 
 class factor extends object_factor_base {
 
+    public function define_add_factor_form_definition($mform) {
+        global $OUTPUT;
```

Comment:
I'm confused about this form, we already know their email. Email should be a zero config factor as the user has already signed up with their email and confirmed it

Reply:
Improved email factor as discussed, so no there is no `add_factor_form` for email factor